File: STRIVER/video/vid-5 (Patterns)/pattern8/p8.cpp
//  * * * * *
//   * * * *
//    * * *
//     * *
//      *

#include <iostream>
using namespace std;

int main()
{
    int row = 3;

    for (int i = 0; i <= row; i++)
    {
        // space
        for (int j = 0; j < i; j++)
        {
            cout << "  ";
        }
        // starts
        for (int j = 0; j < 2 * row - (2 * i + 1); j++)
        {
            cout << "* ";
        }
        // space
        for (int j = 0; j < i; j++)
        {
            cout << "  ";
        }
        cout << endl;
    }
    return 0;
}

File: STRIVER/video/vid-5 (Patterns)/pattern8/p8.py
    #  * * * * *  
    #   * * * *  
    #    * * *   
    #     * *     
    #      *


def p8(row:int)->None:
    for i in range(row+1):
        # space
        for j in range(i):
            print("  ",end="")

        # stars
        for j in range(2*row - (2*i+1)):
            print("* ",end="")

        # space
        for j in range(i):
            print("  ",end="")

        print("")

row = 3
p8(row)